internal/pkg/eql/visitor.go

```diff
@@ -250,6 +250,13 @@ func (v *expVisitor) VisitExpNot(ctx *parser.ExpNotContext) interface{} {
 	return !val
 }
 
+func (v *expVisitor) VisitExpEVariable(ctx *parser.ExpEVariableContext) interface{} {
+	// escaped variables are not allowed in Eql conditions
+	// they should be wrapped in a constant to be allowed as a valid string in the expression
+	v.err = errors.New("escaped variable $${ is not allowed")
```

Comment:
(nitpick) Would this error message be clear in the logs that this is not allowed in a condition?

Reply:
Yes the condition checker adds that context.